cmd/erasure-healfile.go

```diff
@@ -132,24 +139,45 @@ func (s ErasureStorage) HealFile(staleDisks []StorageAPI, volume, path string,
 
 		// write computed shards as chunks on file in each
 		// stale disk
+		writeSucceeded := false
 		for i, disk := range staleDisks {
-			if disk == nil {
+			// skip nil disk or disk that had error on
+			// previous write
+			if disk == nil || writeErrors[i] != nil {
 				continue
 			}
 
 			err = disk.AppendFile(healVolume, healPath, blocks[i])
 			if err != nil {
-				return f, traceError(err)
+				writeErrors[i] = traceError(err)
+			} else {
+				hashers[i].Write(blocks[i])
+				writeSucceeded = true
+			}
+		}
+
+		// If all disks had write errors we quit.
+		if !writeSucceeded {
```

Comment:
Maybe we can refactor this into a separate function?
Small helper function iterating over the error slice and return a combined error if all errors are no-nil?

Reply:
Done.